### PR TITLE
Feature/refactor client

### PR DESCRIPTION
**Pull request by garyluu**
Just refactoring the report command a bit.  Other commands will be done around the time ToolTester is actually used.

### CODE REVIEW

File: tooltester/src/main/java/io/dockstore/tooltester/helper/GA4GHHelper.java [hunk 1]
```diff
@@ -87,4 +91,47 @@ private static boolean matchVerifiedSource(List<String> filter, String verifiedS
...
+     * @param runner             The runners available ("cwltool", "cwl-runner", "cromwell"
+     * @param descriptorTypeEnum The descriptor type of the ToolVersion
+     * @return Whether the runner can be used for the descriptor type
+     */
+    static boolean runnerSupportsDescriptorType(String runner, ToolVersion.DescriptorTypeEnum descriptorTypeEnum) {
+        switch (runner) {
+        case "cwltool":
+        case "cwl-runner":
+            return descriptorTypeEnum == ToolVersion.DescriptorTypeEnum.CWL;
+        case "cromwell":
```
Comment (coverbeck) [inline tooltester/src/main/java/io/dockstore/tooltester/helper/GA4GHHelper.java]: Does the fact that we can run CWL with Cromwell matter here?

File: tooltester/src/main/java/io/dockstore/tooltester/client/cli/ReportCommand.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                    date = TimeHelper.timeFormatConvert(date);
+                } catch (ParseException e) {
+                    errorMessage("Could not parse start time " + date, CLIENT_ERROR);
+                }
+                List<String> record = Arrays
+                        .asList(date, toolVersion.getId(), tag, runner, pipelineNode.getDisplayName(), result, duration, logURL);
+                report.printAndWriteLine(record);
+                if (SEND_LOGS) {
+                    if (result.equals(StateEnum.SUCCESS.name())) {
+                        S3Client s3Client = new S3Client();
```
Comment (coverbeck) [inline tooltester/src/main/java/io/dockstore/tooltester/client/cli/ReportCommand.java]: Is it expensive to create an S3Client? If yes, would it make sense to just create this once outside the loop? If no, then ignore this comment.

File: tooltester/src/main/java/io/dockstore/tooltester/client/cli/ReportCommand.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        String toolVersionId = toolVersion.getId();
+        String tag = toolVersion.getName();
+        String jenkinsJobName = buildName(runner, toolVersionId);
+        if (pipelineTester.getJenkinsJob(jenkinsJobName) == null) {
+            LOG.info("Could not get job: " + jenkinsJobName);
+            return;
+        }
+        int buildId = pipelineTester.getLastBuildId(jenkinsJobName);
+        if (buildId == 0 || buildId == -1) {
+            LOG.info("No build was ran for " + jenkinsJobName);
```
Comment (coverbeck) [inline tooltester/src/main/java/io/dockstore/tooltester/client/cli/ReportCommand.java]: Should be `was run`

File: tooltester/src/main/java/io/dockstore/tooltester/helper/GA4GHHelper.java (hunk 1 above)
Comment (garyluu) [reply tooltester/src/main/java/io/dockstore/tooltester/helper/GA4GHHelper.java]: Currently it's untested

**Pull request merged**